agot-bg-game-server/src/common/ingame-game-state/action-game-state/resolve-march-order-game-state/combat-game-state/post-combat-game-state/PostCombatGameState.ts

```diff
@@ -234,11 +234,8 @@ export default class PostCombatGameState extends GameState<
             // It might be that this movement can be prevented by house cards (e.g. Arianne Martell)
             if (!this.isAttackingArmyMovementPrevented()) {
                 this.combat.resolveMarchOrderGameState.moveUnits(this.combat.attackingRegion, this.combat.attackingArmy, this.combat.defendingRegion);
-            } else {
-                // Defender had to retreat
-                // Therefore possible orders in defending region need to be removed
-                this.removeOrderFromRegion(this.combat.defendingRegion);
             }
+            this.removeOrderFromRegion(this.combat.defendingRegion);
```

Comment:
I think this PR is not necessary. Orders will be removed in `moveUnits()`. The else path is for the case that the movement is prevent and therefore possible orders won't be removed as moveUnits isn't called then.

Reply:
Yup, plus I don't think it will solve the issue. The issue is that in `moveUnits` (that is used both for moving units during a non-combat and a combat march order), it checks if the target destination belongs to the moving army before removeing an order. This is done because in the case of a non-combat march order, you don't remove the order in the target destination if you just moved unit to one your region.

Reply:
Forget my comment. You are right! In case of a user retakes his capital the order will stay...

Reply:
> 
> 
> Yup, plus I don't think it will solve the issue. The issue is that in `moveUnits` (that is used both for moving units during a non-combat and a combat march order), it checks if the target destination belongs to the moving army before removeing an order. This is done because in the case of a non-combat march order, you don't remove the order in the target destination if you just moved unit to one your region.

Indeed... It doesn't solve the problem properly. But @mmasztalerczuk found an issue:

This part of moveUnits must be somehow adapted:

`    moveUnits(from: Region, units: Unit[], to: Region): void {
        const controllerToRegion = to.getController();

        if (controllerToRegion != units[0].allegiance) {...`

In case a user retakes his capital the if path won't be reached as `getController()` will return the original controller (retreat of loser has already been done).